File: easy/147_lettercase_percentage_ratio.py
import sys

from collections import Counter

for line in open(sys.argv[1]):
    c = Counter('l' if c.islower() else 'U' for c in line.strip())
    pc_lowercase = (float(c['l']) / (c['l'] + c['U'])) * 100
    print('lowercase: {:.2f} uppercase: {:.2f}'.format(pc_lowercase, 100-pc_lowercase))


import sys

for line in open(sys.argv[1]):
    uppercase = lowercase = 0
    for c in line.strip():
        if c.islower():
            lowercase += 1
        else:       # assumes no digits, punctuation etc. in the input
            uppercase += 1
    pc_lowercase = (float(lowercase) / (lowercase + uppercase)) * 100
    print('lowercase: {:.2f} uppercase: {:.2f}'.format(pc_lowercase, 100-pc_lowercase))

File: easy/147_lettercase_percentage_ratio.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc, const char * argv[]) {
    FILE *file = fopen(argv[1], "r");
    char line[1024];
    int uppercase, lowercase;
    float pc_lowercase;

    while (fgets(line, 1024, file)) {
        uppercase = lowercase = 0;
        if (line[strlen(line)-1] == '\n')
            line[strlen(line)-1] = '\0';    // get rid of new line

        for (char *p=line; *p; p++) {
            if (islower(*p))
                lowercase++;
            else
                uppercase++;
        }
        pc_lowercase = ((float)lowercase / (lowercase + uppercase)) * 100;
        printf("lowercase: %.2f uppercase: %.2f\n", pc_lowercase, 100-pc_lowercase);
    }

    return 0;
}
